### PR TITLE
Load balancer test module

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
Part of #16

### CODE REVIEW

**Review by lego963 — DISMISSED** [review]

File: playbooks/module_utils/message.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+        try:
+            results = self.run()
+            if results and isinstance(results, dict):
+                self.ansible.exit_json(**results)
+
+        except Exception as e:
+            self.ansible.fail_json(msg=str(e))
+
+    @staticmethod
```
Comment (outcatcher) [inline playbooks/module_utils/message.py:78]: Is there a need for putting this function inside the class?

File: playbooks/module_utils/message.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    def serialize(msg) -> str:
+        """Serialize data as json string"""
+        try:
+            return json.dumps(msg, separators=(',', ':'))
+        except json.JSONDecodeError as err:
+            return err.msg
+
+    @staticmethod
+    def create_metric(name,
```
Comment (outcatcher) [inline playbooks/module_utils/message.py:87]: Same as previous

File: playbooks/library/lb_load_monitoring.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+---
+module: lb_load_monitoring
+short_description: Load balancer response times.
+version_added: "1.0.0"
+author: "Anton Sidelnikov (@anton-sidelnikov)"
+description:
+  - Get metrics from load balancer hosts and push it to unix socket server
+  - APIMON_PROFILER_MESSAGE_SOCKET environment variable should be set.
+options:
+  lb_ip:
```
Comment (outcatcher) [inline playbooks/library/lb_load_monitoring.py]: Maybe should be named `target_address` or something like that?

File: playbooks/library/lb_load_monitoring.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      sample: "metric"
+'''
+
+EXAMPLES = '''
+# Get list of floating IPs statuses (all parameters are specified)
+- lb_load_monitoring:
+    lb_ip: "80.158.53.138"
+  register: out
+'''
+import os
```
Comment (outcatcher) [inline playbooks/library/lb_load_monitoring.py]: Is it necessary to put imports after the documentation?

File: playbooks/library/lb_load_monitoring.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        lb_ip=dict(type='str', required=True),
+        timeout=dict(type='int', default=20)
+    )
+
+    def run(self):
+
+        timeout = self.params['timeout']
+        address = f"http://{self.params['lb_ip']}"
+        metrics = []
+        for _ in range(30):
```
Comment (outcatcher) [inline playbooks/library/lb_load_monitoring.py]: Maybe make request count configurable?

File: playbooks/library/lb_load_monitoring.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+class LbLoadMonitoring(MessageModule):
+    argument_spec = dict(
+        lb_ip=dict(type='str', required=True),
+        timeout=dict(type='int', default=20)
+    )
+
+    def run(self):
+
+        timeout = self.params['timeout']
+        address = f"http://{self.params['lb_ip']}"
```
Comment (outcatcher) [inline playbooks/library/lb_load_monitoring.py]: Let's make the protocol configurable

File: playbooks/module_utils/message.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        if self.ansible._debug or self.ansible._verbosity > 2:
+            self.ansible.log(
+                " ".join(['[DEBUG]', msg]))
+
+    @abc.abstractmethod
+    def run(self):
+        pass
+
+    def __call__(self):
+        """Execute `run` function when calling the class.
```
Comment (outcatcher) [inline playbooks/module_utils/message.py]: Not calling the `class`, but calling an `instance`

File: playbooks/module_utils/message.py [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+    def debug(self, msg):
+        """Prints debug message to system log
+
+        Arguments:
+            msg {str} -- Debug message.
+        """
+        if self.ansible._debug or self.ansible._verbosity > 2:
+            self.ansible.log(
+                " ".join(['[DEBUG]', msg]))
```
Comment (outcatcher) [inline playbooks/module_utils/message.py]: `" ".join(['[DEBUG]', msg]))` can be simplified (?) to `f'[DEBUG] {msg}'`

File: playbooks/module_utils/message.py [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+            'value': value,
+            'environment': environment,
+            'zone': zone
+        }
+        if 'metric_type' in kwargs:
+            message['metric_type'] = kwargs['metric_type']
+        if 'az' in kwargs:
+            message['az'] = kwargs['az']
+        else:
+            message['az'] = 'default'
```
Comment (outcatcher) [inline playbooks/module_utils/message.py]: You can use `kwargs.get(key, default)` instead this `if/else`

File: playbooks/module_utils/message.py [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+            return json.dumps(msg, separators=(',', ':'))
+        except json.JSONDecodeError as err:
+            return err.msg
+
+    @staticmethod
+    def create_metric(name,
+                      value,
+                      environment: str = None,
+                      zone: str = None, **kwargs):
+        """
```
Comment (outcatcher) [inline playbooks/module_utils/message.py:91]: Please add function description to the doc

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**